### ISSUE TITLE
Failure to parse custom registry schemas with referenced types 

### ISSUE THREAD

**Issue by aitormagan** [aws-cloudformation/cfn-lint#3178]
### CloudFormation Lint Version

cfn-lint 0.86.4

### What operating system are you using?

Mac

### Describe the bug

CFN Lint fails to parse custom registry schemas when these schemas include referenced properties.

In our case, we need to validate CloudFormation templates with DataDog Monitors. DataDog Monitors Schema can be found [here](https://github.com/DataDog/datadog-cloudformation-resources/blob/master/datadog-monitors-monitor-handler/datadog-monitors-monitor.json) so we downloaded the schema and executed:

```
cfn-lint cfn_template.yml --registry-schemas <SCHEMAS_FOLDER>
```

where `<SCHEMAS_FOLDER>` is the folder where the schema has been downloaded.

After doing that, we receive the following error:

```
E0002 Unknown exception while processing rule E1010: 'type'
cfn_schema/datadog-monitors-monitor.json:1:1

E0002 Unknown exception while processing rule E1029: 'type'
cfn_schema/datadog-monitors-monitor.json:1:1
```

This error arises because the definition of the schema includes references to data types defined in the `definitions` section. If we modify the schema and "de-refer" the references, this error disappears.

### Expected behavior

We expect `cnf-lint` to work seamlessly with schemas which contains references to data types.  We expect `cfn-lint` to validate DataDog Monitor resources with the definition provided by the provider. 

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  DatadogMonitorLambda1Errors:
    Type: "Datadog::Monitors::Monitor"
    Properties:
      Type: "metric alert"
      Name: "example"
      Query: "sum(last_5m):avg:aws.lambda.errors > 1"
      Message: "example"
      Tags:
        - "tag1"
        - "tag2"
      Options:
        EnableLogsSample: true
        NotifyAudit: false
        Locked: false
        TimeoutH: 0
        NewHostDelay: 300
        RequireFullWindow: false
        RenotifyInterval: 0
        Thresholds:
          Critical: 0
        NotifyNoData: false
        EvaluationDelay: 900
        IncludeTags: true
      Priority: 1
```
